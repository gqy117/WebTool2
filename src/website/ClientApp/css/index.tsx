import "./application.scss";
import "./site.scss";
import "./widget.scss";
