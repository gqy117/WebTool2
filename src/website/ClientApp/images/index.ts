export interface Image {
    health00to19: string;
    health20to39: string;
    health40to59: string;
    health60to79: string;
    health80plus: string;
}

export const Images: Image = {
    health00to19: require("./health-00to19.svg") as string,
    health20to39: require("./health-20to39.svg") as string,
    health40to59: require("./health-40to59.svg") as string,
    health60to79: require("./health-60to79.svg") as string,
    health80plus: require("./health-80plus.svg") as string,
};
