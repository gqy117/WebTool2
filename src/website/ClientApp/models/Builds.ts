export interface BuildConfigration {
    buildId: string;
    refreshInverval?: number ;
}

export interface BuildViewDTO {
    view?: BuildView;
}

export interface BuildView {
    buildId?: string;
    buildStatus?: string;
    buildName?: string;
    lastUpdated?: string;
    cssStatus: string;
    imageSrc: string;
}

export interface BuildState {
    buildState?: Build[];
}

export interface Build {
    buildId?: string;
    buildStatus?: string;
    buildName?: string;
    lastUpdated?: string;
}

export const enum ActionType {
    REQUEST_BUILD = "REQUEST_BUILD",
    RECEIVE_BUILD = "RECEIVE_BUILD",
}

export const enum BuildStatus {
    SUCCESSFUL = "Successful",
    FAILED = "Failed",
}

export const enum CssStatus {
    SUCCESSFUL = "status-successful",
    WARNING = "status-warning",
}

export interface RequestBuildsAction {
    type: ActionType.REQUEST_BUILD;
}

export interface ReceiveBuildsAction {
    type: ActionType.RECEIVE_BUILD;
    build: Build;
}

export type KnownAction = RequestBuildsAction | ReceiveBuildsAction;
