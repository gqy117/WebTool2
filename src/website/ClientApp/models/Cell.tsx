import { RowRenderProps } from "react-table";

export interface CellProps {
    className?: string;
    rowRenderProps: RowRenderProps;
}
