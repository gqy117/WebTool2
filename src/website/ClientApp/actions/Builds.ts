import { fetch, addTask } from "domain-task/index";
import { AppThunkAction } from "../reducers";
import * as Model from "../models/Builds";

export const actionCreators = {
    requestBuilds: (buildId?: string): AppThunkAction<Model.KnownAction> => (dispatch) => {
        let fetchTask = fetch(`api/TeamCity/Build?buildId=${buildId}`)
            .then(response => response.json() as Promise<Model.Build>)
            .then(data => {
                //console.log(data);
                dispatch({ type: Model.ActionType.RECEIVE_BUILD, build: data });
            });

        addTask(fetchTask);
        dispatch({ type: Model.ActionType.REQUEST_BUILD });
    }
};