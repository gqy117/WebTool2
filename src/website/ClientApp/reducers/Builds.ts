import { Reducer } from "redux/index";
import * as Model from "../models/Builds";

const initialState: Model.BuildState = { buildState: [] };

export const reducer: Reducer<Model.BuildState> = (state: Model.BuildState = initialState, action: Model.KnownAction) => {
    switch (action.type) {
        case Model.ActionType.RECEIVE_BUILD:
            const buildWithoutCurrent = (state.buildState as Model.Build[]).filter((x) => x.buildId !== action.build.buildId);

            state = {
                buildState: [
                    ...buildWithoutCurrent,
                    action.build,
                ]
            };
            break;

        case Model.ActionType.REQUEST_BUILD:
        default:
            break;
    }
    // console.log(state);
    return state;
};
