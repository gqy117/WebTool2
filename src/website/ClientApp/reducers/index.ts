import * as BuildModel from "../models/Builds";
import * as ContactModel from "../models/Contacts";
import * as Builds from "./Builds";
import * as Contacts from "./Contacts";
import * as Counter from "./Counter";
import * as WeatherForecasts from "./WeatherForecasts";

// The top-level state object
export interface ApplicationState {
    counter: Counter.CounterState;
    weatherForecasts: WeatherForecasts.WeatherForecastsState;
    builds: BuildModel.BuildState;
    contacts: ContactModel.ContactState;
}

// Whenever an action is dispatched, Redux will update each top-level application state property using
// the reducer with the matching name. It"s important that the names match exactly, and that the reducer
// acts on the corresponding ApplicationState property type.
export const reducers = {
    builds: Builds.reducer,
    contacts: Contacts.reducer,
    counter: Counter.reducer,
    weatherForecasts: WeatherForecasts.reducer,
};

// This type can be used as a hint on action creators so that its "dispatch" and "getState" params are
// correctly typed to match your store.
export type AppThunkAction<TAction> = (dispatch: (action: TAction) => void, getState: () => ApplicationState) => void;
