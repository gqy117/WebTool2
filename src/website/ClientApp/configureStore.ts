import { History } from "history";
import { routerMiddleware, routerReducer } from "react-router-redux";
import { applyMiddleware, combineReducers, compose, createStore, GenericStoreEnhancer, ReducersMapObject, Store, StoreEnhancerStoreCreator } from "redux";
import createSagaMiddleware from "redux-saga";
import thunk from "redux-thunk";
import * as StoreModule from "./reducers";
import { ApplicationState, reducers } from "./reducers";
import rootSaga from "./sagas";

export default function configureStore(history: History, initialState?: ApplicationState) {
    // Build middleware. These are functions that can process the actions before they reach the store.
    const windowIfDefined = typeof window === "undefined" ? null : window as any;

    const sagaMiddleware = createSagaMiddleware();

    // If devTools is installed, connect to it
    const devToolsExtension = windowIfDefined && windowIfDefined.__REDUX_DEVTOOLS_EXTENSION__ as () => GenericStoreEnhancer;
    const createStoreWithMiddleware = compose(
        applyMiddleware(thunk, routerMiddleware(history), sagaMiddleware),
        devToolsExtension ? devToolsExtension() : <S>(next: StoreEnhancerStoreCreator<S>) => next
    )(createStore);

    // Combine all reducers and instantiate the app-wide store instance
    const allReducers = buildRootReducer(reducers);
    const store = createStoreWithMiddleware(allReducers, initialState) as Store<ApplicationState>;

    // then run the saga
    sagaMiddleware.run(rootSaga);

    // Enable Webpack hot module replacement for reducers
    if (module.hot) {
        module.hot.accept("./reducers", () => {
            const nextRootReducer = require<typeof StoreModule>("./reducers");
            store.replaceReducer(buildRootReducer(nextRootReducer.reducers));
        });
    }

    return store;
}

function buildRootReducer(allReducers: ReducersMapObject) {
    return combineReducers<ApplicationState>(Object.assign({}, allReducers, { routing: routerReducer }));
}
