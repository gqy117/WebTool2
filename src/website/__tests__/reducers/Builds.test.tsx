import * as Model from "../../ClientApp/models/Builds";
import { ApplicationState, reducers } from "../../ClientApp/reducers";
import { initialState, reducer } from "../../ClientApp/reducers/Builds";
import * as Const from "../consts/Builds";

describe("reducers/Builds", () => {
    let currentState: Model.BuildState = initialState;

    it("1. Add 'DCE_MAIN_Successful'", () => {
        currentState = reducer(currentState, { type: Model.ActionType.RECEIVE_BUILD, build: Const.DCE_MAIN_SUCCESSFUL });
        expect(currentState).toMatchSnapshot();
    });

    it("2. Add 'DCE_POSTMAN_Successful'", () => {
        currentState = reducer(currentState, { type: Model.ActionType.RECEIVE_BUILD, build: Const.DCE_POSTMAN_SUCCESSFUL });
        expect(currentState).toMatchSnapshot();
    });

    it("3. UPDATE 'DCE_MAIN_Failed'", () => {
        currentState = reducer(currentState, { type: Model.ActionType.RECEIVE_BUILD, build: Const.DCE_MAIN_FAILED });
        expect(currentState).toMatchSnapshot();
    });
});
