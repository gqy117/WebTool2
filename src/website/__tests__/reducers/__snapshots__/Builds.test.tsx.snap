// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/Builds 1. Add 'DCE_MAIN_Successful' 1`] = `
Object {
  "buildState": Array [
    Object {
      "buildId": "DCEMAIN",
      "buildName": "DCE MAIN",
      "buildStatus": "Successful",
      "interval": 60000,
      "lastUpdated": "2017-01-05T18:38:23.2244277+13:00",
    },
  ],
}
`;

exports[`reducers/Builds 2. Add 'DCE_POSTMAN_Successful' 1`] = `
Object {
  "buildState": Array [
    Object {
      "buildId": "DCEMAIN",
      "buildName": "DCE MAIN",
      "buildStatus": "Successful",
      "interval": 60000,
      "lastUpdated": "2017-01-05T18:38:23.2244277+13:00",
    },
    Object {
      "buildId": "DCEPOSTMAN",
      "buildName": "DCE POSTMAN",
      "buildStatus": "Successful",
      "interval": 60000,
      "lastUpdated": "2016-01-05T18:38:23.2244277+13:00",
    },
  ],
}
`;

exports[`reducers/Builds 3. UPDATE 'DCE_MAIN_Failed' 1`] = `
Object {
  "buildState": Array [
    Object {
      "buildId": "DCEPOSTMAN",
      "buildName": "DCE POSTMAN",
      "buildStatus": "Successful",
      "interval": 60000,
      "lastUpdated": "2016-01-05T18:38:23.2244277+13:00",
    },
    Object {
      "buildId": "DCEMAIN",
      "buildName": "DCE MAIN",
      "buildStatus": "Failed",
      "interval": 5000,
      "lastUpdated": "2018-01-05T18:38:23.2244277+13:00",
    },
  ],
}
`;
